homeassistant/components/sentry/__init__.py

```diff
@@ -44,15 +51,98 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
 
     # https://docs.sentry.io/platforms/python/logging/
     sentry_logging = LoggingIntegration(
-        level=logging.INFO,  # Capture info and above as breadcrumbs
+        level=logging.WARNING,  # Capture info and above as breadcrumbs
         event_level=logging.ERROR,  # Send errors as events
     )
 
+    # Find channel based on version number
+    channel = "stable"
+    if "dev0" in current_version:
+        channel = "dev"
+    elif "dev" in current_version:
+        channel = "nightly"
+    elif "b" in current_version:
+        channel = "beta"
+
+    # Additional/extra data collection
+    huuid = await hass.helpers.instance_id.async_get()
+    system_info = await hass.helpers.system_info.async_get_system_info()
+    custom_components = await async_get_custom_components(hass)
+
+    def before_send(event, hint):
+        # Filter out handled events by default
+        if "tags" in event and event.tags.get("handled", "no") == "yes":
+            return None
+
+        # Additional tags to add to the event
+        additional_tags = {
+            "channel": channel,
+            "installation_type": system_info["installation_type"],
+            "uuid": huuid,
+        }
+
+        # Find out all integrations in use, filter "auth", because it
+        # triggers security rules, hiding all data.
+        integrations = [
+            integration
+            for integration in hass.config.components
+            if integration != "auth" and "." not in integration
+        ]
+
+        # Additional extra data to add to the event
+        extra_data = {
+            **system_info,
+            "custom_components": "\n".join(sorted(custom_components.keys())),
```

Comment:
keys is the default iterator of a dict

```suggestion
            "custom_components": "\n".join(sorted(custom_components)),
```